### PR TITLE
feat(ci): test rust by examples (extract to F*)

### PR DESCRIPTION
**Pull request by W95Psp**
Fixes https://github.com/hacspec/hax/issues/540, but instead of trying two snippets, it's trying 157 snippets.

41 are not going though, on a total of 198: we have about 80% success rate here.

This needs #1090.

### CODE REVIEW

File: .utils/rust-by-example.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// This script expect Rust By Example to be in current directory
```
Comment (franziskuskiefer) [inline .utils/rust-by-example.js:1]:
```suggestion
// This script expects Rust By Example to be in current directory
```

Comment (franziskuskiefer) [inline .utils/rust-by-example.js:1]: Please add here where to get it from and how to use this file.

**Review by franziskuskiefer — APPROVED** [review]
lgtm with the comments addressed.

File: .utils/rust-by-example.js (hunk 1 above)
Comment (W95Psp) [reply .utils/rust-by-example.js]: Done!

**Pull request merged**